src/ruby/lib/grpc/generic/active_call.rb

```diff
@@ -226,6 +226,23 @@ def send_status(code = OK, details = '', assert_finished = false,
       nil
     end
 
+    def server_unary_response(req, marshalled: false, trailing_metadata: {},
+                              code: Core::StatusCodes::OK, details: 'OK')
+      @send_initial_md_mutex.synchronize do
+        ops = {}
+        ops[SEND_INITIAL_METADATA] = @metadata_to_send unless @metadata_sent
+        @metadata_sent = true unless @metadata_sent
```

Comment:
You don't need the "unless" here. `@metadata_sent` should be true after this line, one way or the other.